TBCCore/TBCKVOObservation.h

```diff
@@ -0,0 +1,21 @@
+//  Copyright © 2018 Tabcorp Pty. Ltd. All rights reserved.
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^TBCKVOObservationBlock)(NSDictionary<NSKeyValueChangeKey,id> * change);
+
+@interface TBCKVOObservation : NSObject
+
++ (instancetype)observationWithSelector:(SEL)selector object:(NSObject * __nullable)object block:(TBCKVOObservationBlock)block;
+
+- (instancetype)init NS_UNAVAILABLE;
+- (instancetype)initWithSelector:(SEL)selector object:(NSObject * __nullable)object options:(NSKeyValueObservingOptions)options block:(TBCKVOObservationBlock)block __attribute((objc_designated_initializer));
+- (instancetype)initWithSelector:(SEL)selector object:(NSObject * __nullable)object block:(TBCKVOObservationBlock)block;
+
+- (void)removeObservation;
```

Comment:
Not `-invalidate`?

Reply:
Consistency with `TBCNotificationCenterObservation`